### ISSUE TITLE
Unified packcc repository (Was: Are you Arihiro Yoshida?)

### ISSUE THREAD

**Issue by masatake** [arithy/packcc#13]
Some repositories are maintaining and/or developing packcc at GitHub.

* https://github.com/enechaev/packcc
* https://github.com/Leandros/PackCC
* https://github.com/universal-ctags/packcc (I run)
* ...

I'm looking for someone who can merge the repositories into one.
I guess @arithy is Arihiro Yoshida, the person who wrote the original version of packcc.
He is the best person to lead the merging.

**Comment by dolik-rce** [comment]
Hello @masatake,

I'd definitely support merging all of the repositories in one. It would make sense to share the resources to maintain only one instance.

I have found at least 11 independent repositories (not counting forks), but most of them use the original code. Only 4 of them contains any changes, mostly small bugfixes. The only exception is https://github.com/rst256/packcc_lc, who made some bigger changes to adapt it for Lua, but I think we can ignore that, as it is outside of the original scope of PackCC. The rest should be relatively easy to go through and cherry-pick important changes in this repository.

**Comment by arithy** [comment]
Yes, I'm the original PackCC author.
Thanks for your proposal.

It would be nice for PackCC users to avoid confusion regarding which variant is the best for them, as well as for PackCC developers to avoid duplicate efforts.
However, I respect the independent voluntary activities by the variant PackCC developers.
Of course, I strongly hope that they will provide reports or pull requests containing modifications on important issues, but I cannot force them.

**Comment by masatake** [comment]
@enechaev, how do think merging the changes in your repo to this upstream?

@arithy, thank you for replying, and forgive me the impolite issue title.

@arithy, @dolik-rce,  I think the test facility is needed to merge the changes made in various repositories. How do you think about https://github.com/enechaev/packcc/issues/8#issuecomment-754268147 ?

If all of you are o.k., I will write a minimum test case using [bats](https://github.com/bats-core/bats-core), and send it as a pull request?

**Comment by arithy** [comment]
@masatake, no problem.

**Comment by arithy** [comment]
@masatake, thanks for your valuable proposal.
I have been also aware of missing tests.
Please share your test cases with me via a pull request.

**Comment by dolik-rce** [comment]
@masatake: bats based tests are fine. If you prepare the framework I can add tests for unicode and move the https://github.com/enechaev/packcc/issues/8 to this repository.

**Comment by masatake** [comment]
Please, look at #14. I wonder how to enable GitHub Actions in a repository not owned by myself.

**Comment by masatake** [comment]
I would like to close this issue because I got the answer to the original question.

Please, allow me to make some saucy comments.
I would like you to put your name, "Arihiro Yoshida", in your commit logs of packcc and/or to your GitHub page (https://github.com/arithy).
The name on them helps people who have interests in improving packcc know the repository your run is the place to make contributions.

I was looking for a small PEG-based parser generator. I found packcc in Wikipedia and introduced it in Universal Ctags that I have maintained. I didn't want to introduce packcc because its development was not active. However, I could not avoid introducing packcc because packcc was so attractive.

I had improved packcc to fit with Universal Ctags. At that time, it looked to me that https://github.com/enechaev/packcc is the most active fork repo of packcc. I have not had enough capacity to drive the development of packcc, so I have submitted the improvements to @enechaev's repo. (See  https://github.com/enechaev/packcc/pulls?q=is%3Aclose+is%3Apr+author%3A%40me+ ).

If I knew the original author of packcc run's a repository, I would have submitted the pull request to the repository.
"Arihiro Yoshida" written in the original source code was the only hint to reach you. I searched "Arihiro Yoshida" at GitHub however, I cannot find you.  I think this is not special for me. I guess there are people who are in the same situation.

Your name going public helps potential contributors, who want to enjoy improving packcc, find a playground.

@dolik-rce, how do you think what I wrote here? I strongly believe you are enjoying the new playground:-).

Thank you for reading.

**Comment by arithy** [comment]
@masatake, thanks for your advice.
It's a rational reason to show my real name.
I've set up .gitconfig and my profile on GitHub to do so.

**Comment by dolik-rce** [comment]
@masatake:

> @dolik-rce, how do you think what I wrote here? 

Based on my experience in past few weeks @arithy is capable, responsive and kind developer. I can only add that I believe he's a good leader for this project.

> I strongly believe you are enjoying the new playground:-).

Well, I did enjoy improving PackCC. My original intent was to add Kotlin support into [geany](https://github.com/geany/geany) over christmas, but the rabbit hole went much deeper than I expected. But I definitely do not regret that I spent much more time with it, I've learned a lot of new things and also refreshed my C skills (I haven't used it since university :slightly_smiling_face:)
